### PR TITLE
Docs: adding SNP prerequisites to CoCo website

### PR DESCRIPTION
**Pull request by arvindskumar99**
Adding the required prerequisites to run CoCo using SEV-SNP.

### CODE REVIEW

File: hugo.yaml [hunk 1]
```diff
@@ -167,6 +167,7 @@ params:
     # add "hide_readingtime: true" to the page's front matter
     readingtime:
       enable: false
+    showLightDarkModeMenu: true
```
Comment (fitzthum) [inline hugo.yaml:170]: Can you put this in a separate PR

File: content/en/docs/getting-started/prerequisites/hardware/snp.md [hunk 2]
```diff
@@ -8,5 +8,109 @@ tags:
...
 ---
 
-TODO
+## Platform Setup
+
+> [!WARNING]
+>
+> In order to launch SEV or SNP memory encrypted guests, the host must be prepared with a compatible kernel, `6.8.0-rc5-next-20240221-snp-host-cc2568386`. AMD custom changes and required components and repositories will eventually be upstreamed. 
+
+> [Sev-utils](https://github.com/amd/sev-utils/blob/coco-202402240000/docs/snp.md) is an easy way to install the required host kernel, but it will unnecessarily build AMD compatible guest kernel, OVMF, and QEMU components. The additional components can be used with the script utility to test launch and attest a base QEMU SNP guest. However, for the CoCo use case, they are already packaged and delivered with Kata.
```
Comment (fitzthum) [inline content/en/docs/getting-started/prerequisites/hardware/snp.md:17]: The information in this warning is really what the scope of this entire page should be. 

The prerequisites should not include starting up workloads. It should just tell the reader how to setup their machine.

So, what hw is supported in the first place; how to check that the host is valid; Firmware; Kernel, where to get it; etc

If you really want to have a full e2e flow for SNP, you could include that in the examples section, but I would avoid doing this because it will require more maintenance.

**Review by fitzthum — COMMENTED** [review]
I think we can simplify this a lot.

File: hugo.yaml (hunk 1 above)
Comment (arvindskumar99) [reply hugo.yaml:170]: Sure

File: content/en/docs/getting-started/prerequisites/hardware/snp.md (hunk 2 above)
Comment (arvindskumar99) [reply content/en/docs/getting-started/prerequisites/hardware/snp.md]: @fitzthum I removed the warning, and I relabeled the steps to bringing it up as an example. I think having it there is important, because I refer to that when giving instructions on how to validate that the TEE environment is being used. Let me know if you things should be changed in that section.

File: content/en/docs/getting-started/prerequisites/hardware/snp.md [hunk 3]
```diff
@@ -8,5 +8,107 @@ tags:
 - SEV
 ---
 
-TODO
+## Platform Setup
+
+In order to launch SEV or SNP memory encrypted guests, the host must be prepared with a compatible kernel, `6.8.0-rc5-next-20240221-snp-host-cc2568386`. AMD custom changes and required components and repositories will eventually be upstreamed.
```
Comment (fitzthum) [inline content/en/docs/getting-started/prerequisites/hardware/snp.md:13]: This is inaccurate. The an upstream host kernel can be used for SEV. This page is label as SNP instructions anyway so maybe just remove the mention of SEV.

File: content/en/docs/getting-started/prerequisites/hardware/snp.md [hunk 4]
```diff
@@ -8,5 +8,107 @@ tags:
...
 ---
 
-TODO
+## Platform Setup
+
+In order to launch SEV or SNP memory encrypted guests, the host must be prepared with a compatible kernel, `6.8.0-rc5-next-20240221-snp-host-cc2568386`. AMD custom changes and required components and repositories will eventually be upstreamed. 
+
+[Sev-utils](https://github.com/amd/sev-utils/blob/coco-202402240000/docs/snp.md) is an easy way to install the required host kernel, but it will unnecessarily build AMD compatible guest kernel, OVMF, and QEMU components. The additional components can be used with the script utility to test launch and attest a base QEMU SNP guest. However, for the CoCo use case, they are already packaged and delivered with Kata.
+
+Alternatively, refer to the [AMDESE guide](https://github.com/confidential-containers/amdese-amdsev/tree/amd-snp-202402240000?tab=readme-ov-file#prepare-host) to manually build the host kernel and other components.
```
Comment (fitzthum) [inline content/en/docs/getting-started/prerequisites/hardware/snp.md:17]: So what are we telling beginner users to do? You mention that sev-utils but what exactly are they supposed to do with that? You sort of seem to suggest that they shouldn't use it because it installs too much stuff. Should they be using the second option?

File: content/en/docs/getting-started/prerequisites/hardware/snp.md (hunk 2 above)
Comment (fitzthum) [reply content/en/docs/getting-started/prerequisites/hardware/snp.md]: Prerequisites are for things that come before installing the operator. This page should not include any information about installing the operator or running a workload. If you would like to have an e2e example of using SNP, you can put that in the examples section. If you want to show people how to verify that they are inside an enclave, add a section after sample workload, that has a subsection for checking that you are in a SNP enclave (and could have pages for other platforms)

File: content/en/docs/getting-started/prerequisites/hardware/snp.md (hunk 4 above)
Comment (arvindskumar99) [reply content/en/docs/getting-started/prerequisites/hardware/snp.md:17]: I think I worded it wrong. What I had meant is that the coco tagged version of sev-utils cleans up some of the unnecessary things that it installs, but the second option is also a valid way of getting the right kernel.

**Comment by arvindskumar99** [comment]
@fitzthum is this more along the lines of what you're looking for?

**Review by fitzthum — APPROVED** [review]
LGTM. We can iterate on this in the future but this is a good start. Please fix the prose linting test tho.
